.github/workflows/lint.yml

```diff
@@ -40,4 +29,44 @@ jobs:
         run: |
           rustup update --no-self-update nightly
           rustup +nightly component add clippy
-          make clippy
\ No newline at end of file
+          make clippy
+
+  doc:
+    name: doc stable on ubuntu-latest
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@main
+      - name: Build docs
+        run: |
+          rustup update --no-self-update
+          make doc
+
+  version:
+    name: check rust version consistency
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@main
+        with:
+          profile: minimal
+          override: true
+      - name: check rust versions
+        run: ./scripts/check-rust-version.sh
```

Comment:
While we are at it, let's also change `rust-toolchain` to `rust-toolchain.toml` and update the `check-rust-version.sh` script accordingly.

Reply:
I don't quite understand why we're restricting the current toolchain to the MSRV of the workspace. I've read through #267, #272 and the related miden-vm PRs and issues, and discord.

From what I can tell its issues due to using the nightly compiler -- which afaik we're not using?

Reply:
I think the primary motivation is that everyone is using the same toolchain as we've had issues when things suddenly break if a different version is used. As an example, in https://github.com/facebook/winterfell/issues/277 code that worked fine prior to 1.78 started failing with 1.78 onwards (this was due to the bug in the code, but still).

Another example is something like https://github.com/0xPolygonMiden/miden-vm/issues/917, where nothing broke but there was a significant performance degradation when moving from 1.69 to 1.70 (which fixed itself after 1.74).